### PR TITLE
Add support for untyped origination indexing

### LINKED ISSUE #401: An indexer to fetch all originations (untyped)

### PR DESCRIPTION
**Pull request by 0xflo-tez**
Resolves #401

### CODE REVIEW

**Comment by droserasprout** [comment]
Thanks for the contribution! 🎉

Kind of blockers, except missing docs: #442, #431.

**Comment by 0xflo-tez** [comment]
My pleasure ! Happy to contribute to DipDup, this project is awesome 🔥
Is there anything missing or required on my side ? Or anything I can do to help accelerate the integration in a future release?

**Comment by m-kus** [comment]
Hey @0xflo-tez thanks a lot for the PR!

Sorry for the delay, we have discussed internally and decided that in the long term we would need a generic approach for indexing all operations/big map updates/transfers/events without filtering by contracts.
Separate index types (basically index twins) is indeed a good option because it allows us to apply additional config validations (e.g. forbid certain pattern fields) and to have more effective fetchers for this particular scenario.

Circling back to your PR we need you to make several small changes:
- rename `origination` index kind to `operation_unfiltered`
- add mandatory `types` field (without default value) to the index config and allow only `origination` in the validator (since it is the only option implemented rn)
- ideally inherit `OperationIndex` from `OperationUnfilteredIndex` (but we can do that later ourselves)

**Comment by 0xflo-tez** [comment]
Hey @m-kus ! Thank you for your feedback :)
No problem, I'll take care of it, I'll ping you here once this is done, or if I have any question

File: src/dipdup/config.py [hunk 1]
```diff
@@ -908,6 +920,24 @@ def address_filter(self) -> Set[str]:
...
+class OperationUnfilteredIndexConfig(IndexConfig):
+    """Operation index config
+
+    :param kind: always `operation`
+    :param handlers: List of indexer handlers
+    :param first_level: Level to start indexing from
+    :param last_level: Level to stop indexing at (DipDup will terminate at this level)
+    """
+
+    kind: Literal["operation"]
```
Comment (m-kus) [inline src/dipdup/config.py:933]: Should be `operation_unfiltered` right?

Comment (0xflo-tez) [reply src/dipdup/config.py:933]: I was actually wondering if I had to change that or not. I'll update it!

**Comment by 0xflo-tez** [comment]
@m-kus @droserasprout Let me know if you need me to change anything else :)

File: src/dipdup/config.py [hunk 2]
```diff
@@ -908,6 +920,24 @@ def address_filter(self) -> Set[str]:
...
+
+    :param kind: always `operation`
+    :param handlers: List of indexer handlers
+    :param first_level: Level to start indexing from
+    :param last_level: Level to stop indexing at (DipDup will terminate at this level)
+    """
+
+    kind: Literal["operation_unfiltered"]
+    handlers: Tuple[OperationUnfilteredHandlerConfig, ...]
+    types: Tuple[OperationType, ...] = (OperationType.transaction,)
```
Comment (m-kus) [inline src/dipdup/config.py:935]: Should prob be `OperationType.origination` since we support originations atm only

**Comment by 0xflo-tez** [comment]
My apologies, I did not have time to take care of @m-kus latest feedbacks... Thank you for merging and taking care of it !

**Pull request merged**